subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportsContainer.java

```diff
@@ -27,7 +27,7 @@
  * The reporting configuration for the {@link JacocoReport} task.
  */
 @Incubating
-public interface JacocoReportsContainer extends ReportContainer<Report> {
+public interface JacocoReportsContainer extends ReportContainer<ConfigurableReport> {
```

Comment:
If we're going to change `JacocoReportsContainer` we should probably change them all. Even better, get rid of the type parameter to `ReportsContainer` as it doesn't really make sense and doesn't allow evolution (as we've seen here).